### PR TITLE
5919 feature build

### PR DESCRIPTION
**Pull request by pankaj-cgi**
This branch contains new workflow for creating a feature build from a branch. More details of workflow: https://proactionca.ent.cgi.com/confluence/pages/viewpage.action?pageId=100915398#HNSESBCI/CDPipeline-feature-build-hnsesb:

### CODE REVIEW

File: .github/workflows/feature-build-hniesb.yml [hunk 1]
```diff
@@ -1,36 +1,154 @@
...
+      # Upload the images in image repository in tools namespace
+      - name: Push image to registry
+        id: push-image
+        uses: redhat-actions/push-to-registry@v2
+        with:
+          image: ${{ steps.build-docker-hniesb.outputs.image }}
+          tags: ${{ steps.build-docker-hniesb.outputs.tags }}
+          registry: ${{env.IMAGE_REGISTRY}}/${{env.DEFAULT_NAMESPACE}}
+          username: ${{env.IMAGE_REGISTRY_USER}}
+          password: ${{env.OPENSHIFT_TOKEN}}
```
Comment (weskubo-cgi) [inline .github/workflows/feature-build-hniesb.yml:99]: Inconsistent spacing on braces compared to previous lines.

File: .github/workflows/feature-build-hniesb.yml [hunk 2]
```diff
@@ -1,36 +1,154 @@
...
-  build:
+  # This job runs the maven commands to compile code and build package.
+  # This job also creates docker image and push the image to OpenShift repository
+  createHniEsbFeatureBuild:
     # The type of runner that the job will run on
     runs-on: ubuntu-latest
+    # Map a step output to a job output
+    outputs:
+      OPENSHIFT_IMAGE_SHA: ${{ steps.push-image.outputs.digest }}
+      FEATUREBUILD_NAME: ${{ env.FEATUREBUILD_NAME }}
```
Comment (weskubo-cgi) [inline .github/workflows/feature-build-hniesb.yml:44]: Would FEATURE_BUILD_NAME make more sense in terms of readability (where _ equals space).

File: .github/workflows/feature-build-hniesb.yml [hunk 3]
```diff
@@ -1,36 +1,154 @@
...
+          ssh-key: ${{ secrets.GITOPS_C5839F_PRIVATE_KEY }}
+          repository: bcgov-c/tenant-gitops-c5839f
+          ref: main
+      
+
+      - name: Set up Kustomize
+        uses: imranismail/setup-kustomize@v1
+
+      - name: Update image ID and commit change
+        if: ${{success()}}
```
Comment (weskubo-cgi) [inline .github/workflows/feature-build-hniesb.yml:145]: Inconsistent spacing on braces compared to previous lines.

File: .github/workflows/feature-build-hniesb.yml [hunk 4]
```diff
@@ -1,36 +1,154 @@
...
+  # This job updates the Gitops repository with image created in  this workflow. 
+  # Updating the image in Deployment.yml will result in deploying that image in OpenShift
+  Deploy:
+    runs-on: ubuntu-latest
+    needs: [createHniEsbFeatureBuild]
+    
+    steps:
+      - name: Step to move image to dev namespace
+        run: |
+          echo "This step should move the image to repository to test namespace"
```
Comment (weskubo-cgi) [inline .github/workflows/feature-build-hniesb.yml:111]: This should refer to dev.

File: .github/workflows/feature-build-hniesb.yml [hunk 5]
```diff
@@ -1,36 +1,154 @@
...
+# 4. Run the maven install for hnsesb code from input branch name
+# 5. Run the docker file to create image
+# 6. Push the image to OpenShift repository
+# 7. Deploy the image in Dev environment
 
-name: hnsesb-feature-build
+
+name: feature-build-hnsesb
+
+# Build the image and deploy in OpenShift test environment
```
Comment (weskubo-cgi) [inline .github/workflows/feature-build-hniesb.yml:13]: This should refer to dev.

**Review by weskubo-cgi — CHANGES_REQUESTED** [review]

File: .github/workflows/feature-build-hniesb.yml (hunk 1 above)
Comment (pankaj-cgi) [reply .github/workflows/feature-build-hniesb.yml:99]: updated

File: .github/workflows/feature-build-hniesb.yml (hunk 2 above)
Comment (pankaj-cgi) [reply .github/workflows/feature-build-hniesb.yml:44]: updated

File: .github/workflows/feature-build-hniesb.yml (hunk 4 above)
Comment (pankaj-cgi) [reply .github/workflows/feature-build-hniesb.yml:111]: updated

File: .github/workflows/feature-build-hniesb.yml (hunk 5 above)
Comment (pankaj-cgi) [reply .github/workflows/feature-build-hniesb.yml]: updated

File: .github/workflows/feature-build-hniesb.yml (hunk 3 above)
Comment (pankaj-cgi) [reply .github/workflows/feature-build-hniesb.yml:141]: updated

**Review by weskubo-cgi — APPROVED** [review]

**Pull request merged**